### PR TITLE
Validate department and ccc parameters

### LINKED ISSUE #1: Validate user input against metadata
When an API call is made to the `/course/` and `/category/` endpoints, some request parameters be checked against metadata.py to ensure the parameters are valid. In the event of an invalid API call, an error message can be returned along with the appropriate HTTP status code.

For example:

- https://no8.am/course/ABCD can return `{"error": "ABCD is not a valid department"}`
- https://no8.am/category/CCC/ABCD can return `{"error": "ABCD is not a valid CCC requirement"}`

Get started on this issue by looking at the [URL routes in views.py](https://github.com/nowyasimi/no8am/blob/master/no8am/views.py#L70)

### PR DESCRIPTION
**Pull request by ymittal**
Fixed #1 :smile:

### CODE REVIEW

File: no8am/views.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 	find_course_in_department, fetch_section_details, get_user_format_semester, generate_metadata, \
 	CCC_LIST, CREDIT_LIST, DEPARTMENT_LIST, APPLICATION_ROOT, STATIC_LOCATION, SIMPLE_FORM_TOKEN
 
+from no8am.utility import is_valid_department, is_valid_ccc_req
```
Comment (nowyasimi) [inline no8am/views.py]: Move these imports to the bottom of `__init__.py`

File: no8am/views.py [hunk 2]
```diff
@@ -9,6 +9,9 @@
 	find_course_in_department, fetch_section_details, get_user_format_semester, generate_metadata, \
 	CCC_LIST, CREDIT_LIST, DEPARTMENT_LIST, APPLICATION_ROOT, STATIC_LOCATION, SIMPLE_FORM_TOKEN
```
Comment (nowyasimi) [inline no8am/views.py]: add `is_valid_department, is_valid_ccc_req` to these imports

File: no8am/utility.py [hunk 3]
```diff
@@ -57,6 +57,34 @@ def get_user_format_semester():
...
+        for dept_dic in DEPARTMENT_LIST:
+                if dept_dic["abbreviation"] == dept:
+                        return True
+        return False
+
+
+def is_valid_ccc_req(req):
+        """
+        Checks whether a requirement code is in the list of valid CCC
+        requiremenets.
```
Comment (nowyasimi) [inline no8am/utility.py]: typo in requirements

**Comment by nowyasimi** [comment]
Also, don't worry about the failed automated tests. We need to update the Travis CI settings (#5). The tests are passing locally!

**Comment by ymittal** [comment]
Committed fixes. Ready to merge? 😄

**Comment by nowyasimi** [comment]
Done!

**Pull request merged**